### PR TITLE
Maxgamill sheffield/#300 jpk

### LINKED ISSUE #300: Support JPK formats
As we work towards supporting more fileformats (see #137) we have decided to prioritise being able to load and process JPK file formats. The [afmformats](https://github.com/AFM-analysis/afmformats/) package supports this format and so the `topostats.io` sub-module should be expanded to support this.

Currently loading of Bruker formats is done via `topostats.io.load_scan()` to avoid breaking functionality this should be converted to (another!) Factory Method that calls the appropriate routine for loading the given file format.

**Describe the solution you'd like**

Be able to load JPK files.

**Describe alternatives you've considered**
N/A

### PR DESCRIPTION
**Pull request by MaxGamill-Sheffield**
Closes #300

This PR builds off the `.asd` branch due to the change in input affecting how these files are processed, so it might be best for that to be merged first.

As images are in a `TIFF` format, they are loaded using `tifffile` which comes installed with `scikit-image`.

The jpk file specification document is used to translate the hexadecimal tiff tags and obtain the values needed to process this filetype. These are the: channel, trace/retrace, real and px lengths, and the image.

Tests have also been made for this filetype excluding the px_to_nm scaling function, similar to the other filetypes as this needs some thought.

### CODE REVIEW

**Comment by MaxGamill-Sheffield** [comment]
I have now removed asd and libasd references so this should be able to be merged with dev now.

**Review by ns-rse — COMMENTED** [review]
Some notes on sections that need changing in light of `libasd` now being available as source.

File: setup.cfg
Comment (MaxGamill-Sheffield) [reply setup.cfg:41]: I've instead removed this as it's not a requirement for `jpk` files. But I'll make sure to do this for the asd PR

File: topostats/validation.py
Comment (ns-rse) [inline topostats/validation.py:34]: May have been dealt with in the other PR of .asd support but this can now simply read...

```suggestion
                error="Invalid value in config for 'file_ext', valid values are '.spm', '.jpk', '.ibw' or '.asd'.",
```

File: topostats/run_topostats.py
Comment (ns-rse) [inline topostats/run_topostats.py:141]: Appear to have lost the docstring for the parameter `base_dir`.

File: topostats/io.py
Comment (ns-rse) [inline topostats/io.py:300]: Two debugging lines that can be removed.

File: topostats/io.py
Comment (ns-rse) [inline topostats/io.py:273]: Will these values _always_ be as stated or is there scope for them changing and needing to be configurable options?

File: topostats/io.py
Comment (ns-rse) [inline topostats/io.py:242]: Should really use `LOGGER.error()` rather than `print()`.

File: topostats/io.py
Comment (ns-rse) [inline topostats/io.py:188]: Missed this previously but is there a type for the `ibw` scan?

File: topostats/io.py
Comment (ns-rse) [inline topostats/io.py:129]: Similarly I missed it in previous code reviews but is there a type for the `channel_data`?

File: topostats/io.py
Comment (MaxGamill-Sheffield) [reply topostats/io.py:273]: Yep this should always be the same according to the jpk specification

**Comment by MaxGamill-Sheffield** [comment]
@ns-rse changes addressed. If we want to wait until a few days after the conference to merge to make sure theres a super stable release for people wanting to test things I'm happy for that. 

All should be good but still...

**Review by ns-rse — CHANGES_REQUESTED** [review]
Thanks for finding the time to go through this again @MaxGamill-Sheffield 

I think it just needs the `tifffile` including in the `install_requires` section of `setup.cfg` and should be good to merge.

File: topostats/io.py
Comment (ns-rse) [reply topostats/io.py:10]: Although is `tifffile` a dependency of another package perhaps? :thinking:

Comment (MaxGamill-Sheffield) [reply topostats/io.py:10]: Yep `tiffile` already comes with `scikit-image` as seen from the output of `pip show scikit-image` below:
```
(smth) Max:TopoStats Maxgamill$ pip show scikit-image
Name: scikit-image
Version: 0.19.2
Summary: Image processing in Python
Home-page: https://scikit-image.org
Author: 
Author-email: 
License: Modified BSD
Location: /opt/miniconda3/envs/smth/lib/python3.10/site-packages
Requires: imageio, networkx, numpy, packaging, pillow, PyWavelets, scipy, tifffile
Required-by: pySPM, topostats
```

File: tests/test_io.py
Comment (SylviaWhittle) [inline tests/test_io.py:89]: Should this test be completed before merging?

**Review by ns-rse — APPROVED** [review]
Looks good and happy for this to be merged.

I haven't found any advice on being explicit about dependencies of dependencies. Will investigate further and deal with subsequently.

File: tests/test_io.py
Comment (MaxGamill-Sheffield) [reply tests/test_io.py]: Good catch! It can be removed as its parameterised in the tests below. I think it slipped through the net when merging branches due to it previously being a place holder.

**Comment by MaxGamill-Sheffield** [comment]
@ns-rse I've added the `tifffile` import to save you making an issue, branch changes etc for this. Will merge once its ready

**Pull request merged**
